crates/searchers/openapi.yml

```diff
@@ -0,0 +1,653 @@
+          $ref: "#/components/schemas/Token"
+        amount:
+          $ref: "#/components/schemas/TokenAmount"
+
+    OrderUid:
+      description: |
+        Unique identifier for the order. Order UIDs are 56 bytes long, where
+        bytes [0, 32) represent the order digest used for signing, bytes
+        [32, 52) represent the owner address and bytes [52, 56) represent the
+        order's `validTo` field.
+      type: string
+      example: "0x30cff40d9f60caa68a37f0ee73253ad6ad72b45580c945fe3ab67596476937197854163b1b0d24e77dca702b97b5cc33e0f83dcb626122a6"
+
+    OrderSide:
+      description: |
+        The trading side of the order.
+      type: string
+      enum: [sell, buy]
+
+    OrderClass:
+      description: |
+        How the CoW Protocol order was classified.
+      type: string
+      enum: [market, limit, liquidity]
+
+    AppData:
+      description: |
+        32 bytes of arbitrary application specific data that can be added to an
+        order. This can also be used to ensure uniqueness between two orders
+        with otherwise the exact same parameters.
+      example: "0x0000000000000000000000000000000000000000000000000000000000000000"
+
+    SellTokenBalance:
+      description: |
+        Where should the sell token be drawn from?
+      type: string
+      enum: [erc20, internal, external]
+
+    BuyTokenBalance:
+      description: |
+        Where should the buy token be transfered to?
+      type: string
+      enum: [erc20, internal]
+
+    SigningScheme:
+      description: |
+        How was the order signed?
+      type: string
+      enum: [eip712, ethsign, presign, eip1271]
+
+    Signature:
+      description: |
+        Signature bytes.
+      type: string
+      example: "0x0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000"
+
+    Order:
+      description: |
+        CoW Protocol order information relevant to execution.
+      type: object
+      required:
+        - uid
+        - sellToken
+        - buyToken
+        - sellAmount
+        - buyAmount
+        - feeAmount
+        - side
+        - partiallyFillable
+        - kind
+        - reward
+      properties:
+        uid:
+          $ref: "#/components/schemas/OrderUid"
+        sellToken:
+          $ref: "#/components/schemas/Token"
+        buyToken:
+          $ref: "#/components/schemas/Token"
+        sellAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        buyAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        feeAmount:
+          $ref: "#/components/schemas/TokenAmount"
+        side:
+          $ref: "#/components/schemas/OrderSide"
+        partiallyFillable:
+          description: |
+            Whether or not this order can be partially filled. If this is false,
+            then the order is a "fill-or-kill" order, meaning it needs to be
+            completely filled or not at all.
+          type: boolean
```

Comment:
If we want to support partially fillable orders in this API now then we also need to decide how to communicate how much the order has already been executed and how much balance the user has available. This might be somewhat complicated so not sure if we want to spend time discussing it now.

Reply:
I always imagined that we would just scale the amounts. I think that is what we currently do.